src/main/java/net/glowstone/constants/GlowStatistic.java

```diff
@@ -127,12 +54,111 @@ private GlowStatistic() {
+            return SUB_STATISTICS.get(stat).get(material);
+        }
+
+        return null;
+    }
+
+    /**
+     * Get the statistic name for a specified Statistic and EntityType.
+     *
+     * @param stat the Statistic
+     * @param entityType the EntityType
+     * @return the statistic name
+     */
+    public static String getName(Statistic stat, EntityType entityType) {
+        checkNotNull(stat, "Statistic cannot be null");
+        checkNotNull(entityType, "EntityType cannot be null");
+
+        if (SUB_STATISTICS.containsKey(stat)) {
+            return SUB_STATISTICS.get(stat).get(entityType);
+        }
+
+        return null;
     }
 
-    private static void set(Statistic stat, String key) {
-        names[stat.ordinal()] = "stat." + key;
+    private static void set(Statistic statistic, Enum data, String key) {
+        if (!SUB_STATISTICS.containsKey(statistic)) {
+            SUB_STATISTICS.put(statistic, new HashMap<>());
+        }
+        SUB_STATISTICS.get(statistic).put(data, "stat." + key);
     }
 
+
+    private static void generateStatistics() {
+        for (Statistic stat : values()) {
+            STATISTICS.put(stat, "stat." + UPPER_UNDERSCORE.to(LOWER_CAMEL, stat.name()));
+        }
+
+        // Specific case
+        STATISTICS.put(PLAY_ONE_TICK, "stat.playOneMinute");
+    }
+
+    private static void generateMaterialStatistics() {
+        for (Material material : Material.values()) {
+            String name = material.name().toLowerCase()
+                    .replace("spade", "shovel")
+                    .replace("wood", "wooden");
+
+            if (material != GOLD_INGOT
+                    && material != GOLD_NUGGET
+                    && material != GOLD_BLOCK
+                    && material != GOLD_ORE) {
+                name = name.replace("gold", "golden");
+            }
+
+            if (material.isBlock()) {
+                set(MINE_BLOCK, material, "mineBlock.minecraft." + name);
+            }
+
+            if (material.isItem()) {
+                set(USE_ITEM, material, "useItem.minecraft." + name);
+                set(CRAFT_ITEM, material, "craftItem.minecraft." + name);
+
+                if (material.getMaxDurability() != 0) {
+                    set(BREAK_ITEM, material, "breakItem.minecraft." + name);
+                }
+            }
+
+            set(DROP, material, "drop.minecraft." + name);
+            set(PICKUP, material, "pickup.minecraft." + name);
+        }
+    }
+
+    private static void generateEntityStatistics() {
+        for (EntityType entityType : EntityType.values()) {
+            Class<? extends Entity> entityClass = entityType.getEntityClass();
+
+            if (entityClass == null) {
+                continue;
+            }
+
+            if (Monster.class.isAssignableFrom(entityClass)) {
+                set(ENTITY_KILLED_BY, entityType, "entityKilledBy."
+                        + UPPER_UNDERSCORE.to(UPPER_CAMEL, entityType.name()));
```

Comment:
Why not use `EntityType#getName()`?

Reply:
I believe the deprecation is meant to warn plugin developers about 1.13. We use this method elsewhere in Glowstone and we are still targetting 1.12.

Reply:
I modified using `getName` but some cases are not working. For example, I have ZombiePigman instead of PigZombie. I can add the specific cases after the for loop just like I did for the `PLAY_ONE_TICK`. As you wish

Reply:
I think the SummonCommand implementation is correct, it's just that the stat IDs do not match the Vanilla ones for the entities. Basically, it's just the entity class name. I might be wrong but it's working fine as it is.
I might change my implementation to avoid the `UPPER_UNDERSCORE.to(UPPER_CAMEL, entityType.name())` and switch to just `entityClass.getSimpleName()`